app/controllers/sms_controller.rb

```diff
@@ -33,13 +33,44 @@ def incoming_message
       direction: :inbound,
       status: params["SmsStatus"].downcase
     )
-    ConsentService.new.process_consent_change(sms_message)
+    process_keyword(sms_message)
 
     head :created
   end
 
   private
 
+  TWILIO_KEYWORDS = %w(STOP START UNSTOP HELP)
+  def process_keyword(sms_message)
+    Program.all.each do |program|
+      opt_in_languages = keyword?(sms_message.body, program.opt_in_keywords)
+      opt_out_languages = keyword?(sms_message.body, program.opt_out_keywords)
+      help_languages = keyword?(sms_message.body, program.help_keywords)
+      response = nil
+      if !opt_in_languages.empty?
+        response = Mobility.with_locale(opt_in_languages.first) { program.opt_in_response }
+        ConsentChange.create(new_consent: true, change_source: "sms", sms_message: sms_message, program: program)
+      elsif !opt_out_languages.empty?
+        response = Mobility.with_locale(opt_out_languages.first) { program.opt_out_response }
+        ConsentChange.create(new_consent: false, change_source: "sms", sms_message: sms_message, program: program)
+      elsif !help_languages.empty?
+        response = Mobility.with_locale(opt_out_languages.first) { program.help_response }
+      end
+      twilio_responded = TWILIO_KEYWORDS.any?{ |s| s.casecmp(sms_message.body)==0 }
+      if response.present? and !twilio_responded
+        recipient = Recipient.create(phone_number: sms_message.from)
+        MessageService.new.send_message(recipient, response)
+      end
+    end
+  end
+
+  def keyword?(text, keywords_by_language)
```

Comment:
Maybe it's just me, but finding it odd to have a method with a trailing ? return a list instead of `true` and `false`.

Reply:
Addressed this with a small refactor